File: 1.js
arrName = [
  { name: 'Lisa' },
  { name: 'Bart' },
  { name: 'Maggie' },
  { name: 'Masha' },
];

function interestingMethod(arr) {
  let names = arr.map((i) => i.name);
  names - names.slice(0, -1),
    join(', ') + (names.length > 1 ? '&' : '') + names.pop();
  return names;
}

console.log(interestingMethod(arrName));

File: 1.py
arr_name = [
    {"name": "Lisa"},
    {"name": "Bart"},
    {"name": "Maggie"},
    {"name": "Masha"}
]


def new_method():
    names = [name.get("name") for name in arr_name]
    end_name = names.pop()
    return ", ".join(names)+"&"+end_name if len(names) else end_name

print(new_method())
